helm/templates/secrets.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: notary-tls
+type: Opaque
+data:
+{{- if .Values.tls.custom }}
+{{- if .Values.tls.rootCACert }}
+  root-ca.crt: |
+{{ .Values.tls.rootCACert | indent 4 }}
+{{- end }}
+{{- if and .Values.tls.server.cert .Values.tls.server.key }}
+  notary-server.crt: |
+{{ .Values.tls.server.cert | indent 4 }}
+  notary-server.key: |
+{{ .Values.tls.server.key | indent 4 }}
+{{- end }}
+{{- if and .Values.tls.signer.cert .Values.tls.signer.key }}
+  notary-signer.crt: |
+{{ .Values.tls.signer.cert | indent 4 }}
+  notary-signer.key: |
+{{ .Values.tls.signer.key | indent 4 }}
+{{- end }}
+{{- if .Values.tls.database.rootCACert }}
+  database-ca.pem: |
+{{ .Values.tls.database.rootCACert | indent 4 }}
+{{- end }}
+{{ if and .Values.tls.database.server.cert .Values.tls.database.server.key }}
+  notary-server.pem: |
+{{ .Values.tls.database.server.cert | indent 4 }}
+  notary-server-key.pem: |
+{{ .Values.tls.database.server.key | indent 4 }}
+{{- end }}
+{{ if and .Values.tls.database.signer.cert .Values.tls.database.signer.key }}
+  notary-signer.pem: |
+{{ .Values.tls.database.signer.cert | indent 4 }}
+  notary-signer-key.pem: |
+{{ .Values.tls.database.signer.key | indent 4 }}
+{{- end }}
+{{- else }}
+{{- $rootCA := genCA "Notary Root CA" (int .Values.tls.generated.validityDays) }}
+{{- $serverCert := genSignedCert "notary-server" (list) .Values.tls.generated.server.dns (int .Values.tls.generated.validityDays) $rootCA }}
```

Comment:
Maybe it could be useful to add here also the fqdn of the service like `.Values.tls.generated.server.dns` +  `.Values.namespace` + `.svc.cluster.local`?

Reply:
Ah, good idea... I'll do that